### PR TITLE
Adding Show default amount of rooms button

### PR DESCRIPTION
**Pull request by chagai95**
This should help with https://github.com/vector-im/element-web/issues/16665

image

Signed-off-by: Chagai Friedlander chagai95@gmail.com

Preview: https://619bbbec5be11726eb1f7655--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Adding Show default amount of rooms button ([\#6519](https://github.com/matrix-org/matrix-react-sdk/pull/6519)). Contributed by @chagai95.

### CODE REVIEW

File: src/components/views/rooms/RoomSublist.tsx
Comment (gsouquet) [inline src/components/views/rooms/RoomSublist.tsx:594]: Let's use CSS for styling rather than using semantic HTML elements

File: src/components/views/rooms/RoomSublist.tsx
Comment (gsouquet) [inline src/components/views/rooms/RoomSublist.tsx:601]: That label brings an element of surprise in my opinion. A user might not be really aware of what the default amount of rooms is. At the moment it seems that this number is `8` and set in `ListLayout#L85`.

The question would also be, how will this behave if a user has less than 8 rooms in a given section?

**Review by gsouquet — CHANGES_REQUESTED** [review]

File: src/components/views/rooms/RoomSublist.tsx
Comment (chagai95) [reply src/components/views/rooms/RoomSublist.tsx:594]: Thanks for reviewing! Do you have an example for me? I'd gladly change that

File: src/components/views/rooms/RoomSublist.tsx
Comment (chagai95) [reply src/components/views/rooms/RoomSublist.tsx:601]: I originally wrote 8 but that looked weird so I removed that... Is this blocking the merge? Can we not just iterate on that depending on user feedback? 🙈 I could also write 8 in brackets?

When there are less rooms then nothing happens, I guess it would be better to remove the button in that case, is this also a blocker for the merge?

File: src/components/views/rooms/RoomSublist.tsx
Comment (gsouquet) [reply src/components/views/rooms/RoomSublist.tsx:594]: I would remove the two `<br />` and edit `_RoomSublist.scss` to create the spacing here using `margin-top` or any other relevant CSS property that'll create the effect you're looking for

File: src/components/views/rooms/RoomSublist.tsx
Comment (gsouquet) [reply src/components/views/rooms/RoomSublist.tsx:601]: Well I'm not entirely sure, I'd rather wait for a design review in here.
To me the wording brings an element of surprise, users do not entirely know what will happen until they click that button given that the default is not explicit

Also having the button do nothing when clicked seems that it would confuse user even more.

Not definitevely blocking, but let's wait for a design review and that should clear things up

Comment (chagai95) [reply src/components/views/rooms/RoomSublist.tsx:601]: @gsouquet thanks for replying so fast.

What if I make it a label to start with? Does that make sense? 

Maybe we can have more text explaining in the label tab?

How about adding an explanation when hovering over the button?

File: src/components/views/rooms/RoomSublist.tsx
Comment (chagai95) [reply src/components/views/rooms/RoomSublist.tsx:594]: Sounds good, I think changing Rooms u list will change all elements and I just needed space between the two check boxes and the bottom - they are all part of the list I think, or maybe I'll just remove the button from the list...

**Comment by janogarcia** [comment]
@chagai95 Hi Chagai. I appreciate you taking the time to document that interaction issue and then come up with a proposal, but I'm afraid this is not the best way to solve it.

First, the CTA button makes the whole popover look like a form, for which the "Show default amount of room" would be the submit button.

There's also a clear disconnect between where you interact with the "Show more" action and where you can override its effect. It's hard to discover that connection/relationship. Then, it's also going to be confusing for people seeing such a prominent action by default, for no special reason.

A floating/sticky "Collapse List" action fixed to the bottom of the scrollable area of the room list could solve it in a more intuitive way, but this does require more product/design thinking  and exploration to come up with an elegant solution that works intuitively for everyone.

**Comment by chagai95** [comment]
> @chagai95 Hi Chagai. I appreciate you taking the time to document that interaction issue and then come up with a proposal, but I'm afraid this is not the best way to solve it.

Thanks for answering. Is it ok as a workaround in the labs section?

> 
> First, the CTA button makes the whole popover look like a form, for which the "Show default amount of room" would be the submit button.

If I made it white, will that be better?

> 
> There's also a clear disconnect between where you interact with the "Show more" action and where you can override its effect. It's hard to discover that connection/relationship. Then, it's also going to be confusing for people seeing such a prominent action by default, for no special reason.

I understand you're afraid of confusing people but I cannot find the show less button the way it is done now so the proposal is not meant to be perfect, just a little bit better than what it is now, do you think it's worse than how it is now? I understand it'll confuse people but as a lab feature with an extensive explanation that should be ok then?

> 
> A floating/sticky "Collapse List" action fixed to the bottom of the scrollable area of the room list could solve it in a more intuitive way, but this does require more product/design thinking and exploration to come up with an elegant solution that works intuitively for everyone.

I'm sorry but I'm not following you... Could you elaborate or maybe give an example? A college at work came up with an idea today, why not add another small section similar to the spaces section and have filtering between sections there rather than having to scroll up and down to find the sections... An example for this is Microsoft Teams, if you'd like I can make a screen recording showing this 😉

**Comment by t3chguy** [comment]
> Is it ok as a workaround in the labs section?

Labs is intended for work-in-progress things, rather than workarounds.

**Comment by SimonBrandner** [comment]
> > Is it ok as a workaround in the labs section?
> 
> Labs is intended for work-in-progress things, rather than workarounds.

(or dev tools - hidden events)

**Comment by t3chguy** [comment]
> (or dev tools - hidden events)

Honestly, that shouldn't be a labs flag, should be a toggle in /devtools

**Comment by chagai95** [comment]
> > Is it ok as a workaround in the labs section?
> 
> Labs is intended for work-in-progress things, rather than workarounds.

Ok, so I guess I'm back to asking if it's better to have people confused by this than to have them scroll around forever... Maybe I'll make a video showing how you end up scrolling around... 🙈 I'm open to any other suggestions, I was bothered enough by this to actually create a PR and I'm sure a lot of people are as well...

**Comment by chagai95** [comment]
Sorry for the volume, I'll try making a new video but maybe you can hear something...

https://chagai.website/scrolling-around-to-find-show-less.mp4

**Comment by gsouquet** [comment]
It looks like there's a misalignment between you and the design team at the moment

The best forum to realign and move this forward would be to discuss this over [#matrix-design:matrix.org](https://matrix.to/#/#matrix-design:matrix.org) and then make the code adjustment as I can see a difference in approach here

**Comment by SimonBrandner** [comment]
> It looks like there's a misalignment between you and the design team at the moment
> 
> The best forum to realign and move this forward would be to discuss this over [#matrix-design:matrix.org](https://matrix.to/#/#matrix-design:matrix.org) and then make the code adjustment as I can see a difference in approach here

https://matrix.to/#/#element-design:matrix.org seems more appropriate

**Comment by nadonomy** [comment]
Hey @chagai95 thanks for the PR!

Counter intuitive to the comments above it'd be better to keep the discussion to this PR— we can explore some options and discuss asynchronously, and then discuss sync either in that room or by 1:1 in DM at a point which would be more productive.

We have context in this PR, @janogarcia (designer) from the core team is already context switched into the problem space, but we can't guarantee his availability to discuss this at the time you're posting (he deserves a weekend :).

**Comment by SimonBrandner** [comment]
> Would it make sense to merge the conflicts? I think this problem is not going to be solved soon in a better way, so why not do this? Is the design the issue?

Yes, it is a design issue - you'd need design to approve this change

**Comment by chagai95** [comment]
Ok thx so I won't merge yet, could i get someone's attention on this to decide what design would be ok? If I would add a config for this would it mean i could skip the design team approval?

**Comment by SimonBrandner** [comment]
> Ok thx so I won't merge yet, could i get someone's attention on this to decide what design would be ok? If I would add a config for this would it mean i could skip the design team approval?

The only way to skip design might be a labs option

**Comment by chagai95** [comment]
So a regular config?

**Comment by janogarcia** [comment]
> A floating/sticky "Collapse List" action fixed to the bottom of the scrollable area of the room list could solve it in a more intuitive way, but this does require more product/design thinking and exploration to come up with an elegant solution that works intuitively for everyone.

@chagai95 

So, I've put together some mockups to illustrate what I had in mind when I posted that comment above. I still don't think moving the collapse behavior to the popover menu is the most intuitive thing. 

Basically, a floating action for collapsing room/people lists when those are is fully uncollapsed.

I need to discuss this direction further with the design team (planning it for this Thursday) and make sure it works in dark and narrow viewport modes.

Candidate_ 24px Collapse FAB

**Comment by chagai95** [comment]
Interesting design thx for pinging!

**Review by janogarcia — CHANGES_REQUESTED** [review]
@chagai95 We've discussed the [design direction above](https://github.com/matrix-org/matrix-react-sdk/pull/6519#issuecomment-1195627623) with the team today. Would you be interested in/willing to make the required changes if I provide you with a detailed spec?

**Comment by chagai95** [comment]
What do you mean with a detailed spec? I can definitely try to do something, but I don't want to commit to anything...

**Comment by janogarcia** [comment]
@chagai95 Makes sense. It would be about describing how the "Show less" floating action should behave. That is, details such as when and how it's revealed and hidden.

**Comment by weeman1337** [comment]
@chagai95 I am going to close this PR. Reading the comments the current approach does not align with what design/product wants to have. Nevertheless thank you for your work on this :+1: 

Any further discussion should happen in https://github.com/vector-im/element-web/issues/16665

**Comment by chagai95** [comment]
Ah such a pity, it's still an every day struggle for me....

**Comment by weeman1337** [comment]
> Ah such a pity, it's still an every day struggle for me....

The room list is currently being redesigned. Hopefully the new version will solve your problem. However, nothing is ready yet.

**Pull request closed without merging**
